include/aspect/utilities.h

```diff
@@ -1232,6 +1232,19 @@ namespace aspect
                               const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
                               const double SPD_safety_factor);
 
+    /**
+     * Computes the current_edot_ii and current_stress, with the option for
+     * viscous and viscoelastic contribution.
+     */
+    template <int dim>
+    double compute_current_edot_ii (const unsigned int j,  // the volume fraction
+                                                       const std::vector<double> &composition,
```

Comment:
It's an array, so a plural variable name would be more appropriate. Maybe
```suggestion
                                                       const std::vector<double> &compositions,
```
?